Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm

```diff
@@ -376,52 +365,82 @@ static void enableNotifications(WKWebViewConfiguration *configuration)
 protected:
     WebPushDTest(LaunchOnlyOnce launchOnlyOnce = LaunchOnlyOnce::Yes)
     {
-        [WKWebsiteDataStore _allowWebsiteDataRecordsForAllOrigins];
-
+        m_origin = "https://example.com"_s;
+        m_url = adoptNS([[NSURL alloc] initWithString:@"https://example.com/"]);
         m_tempDirectory = retainPtr(setUpTestWebPushD(launchOnlyOnce));
-
-        auto dataStoreConfiguration = adoptNS([_WKWebsiteDataStoreConfiguration new]);
-        dataStoreConfiguration.get().webPushMachServiceName = @"org.webkit.webpushtestdaemon.service";
-        dataStoreConfiguration.get().webPushDaemonUsesMockBundlesForTesting = YES;
-        m_dataStore = adoptNS([[WKWebsiteDataStore alloc] _initWithConfiguration:dataStoreConfiguration.get()]);
-
-        m_configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
-        m_configuration.get().websiteDataStore = m_dataStore.get();
-        clearWebsiteDataStore([m_configuration websiteDataStore]);
-
-        [m_configuration.get().preferences _setPushAPIEnabled:YES];
-        enableNotifications(m_configuration.get());
-
         m_testMessageHandler = adoptNS([[TestMessageHandler alloc] init]);
-        [[m_configuration userContentController] addScriptMessageHandler:m_testMessageHandler.get() name:@"test"];
-
         m_notificationMessageHandler = adoptNS([[NotificationScriptMessageHandler alloc] init]);
-        [[m_configuration userContentController] addScriptMessageHandler:m_notificationMessageHandler.get() name:@"note"];
     }
 
     void loadRequest(ASCIILiteral htmlSource, ASCIILiteral serviceWorkerScriptSource)
     {
+        ASSERT_FALSE(m_server);
         m_server.reset(new TestWebKitAPI::HTTPServer({
             { "/"_s, { htmlSource } },
             { "/constants.js"_s, { { { "Content-Type"_s, "application/javascript"_s } }, constants } },
             { "/sw.js"_s, { { { "Content-Type"_s, "application/javascript"_s } }, serviceWorkerScriptSource } }
-        }, TestWebKitAPI::HTTPServer::Protocol::Http));
+        }, TestWebKitAPI::HTTPServer::Protocol::HttpsProxy));
+
+        auto dataStoreConfiguration = adoptNS([_WKWebsiteDataStoreConfiguration new]);
+        [dataStoreConfiguration setProxyConfiguration:@{
+            (NSString *)kCFStreamPropertyHTTPSProxyHost: @"127.0.0.1",
+            (NSString *)kCFStreamPropertyHTTPSProxyPort: @(m_server->port())
+        }];
+        dataStoreConfiguration.get().webPushMachServiceName = @"org.webkit.webpushtestdaemon.service";
+        dataStoreConfiguration.get().webPushDaemonUsesMockBundlesForTesting = YES;
+
+        // FIXME: This seems like it shouldn't be necessary, but _clearResourceLoadStatistics (called by clearWebsiteDataStore) doesn't seem to work.
+        NSError *error = nil;
+        [[NSFileManager defaultManager] removeItemAtURL:[dataStoreConfiguration _resourceLoadStatisticsDirectory] error:&error];
+        ASSERT_NULL(error);
```

Comment:
Seems like this is why the ITPCleanup test failed. To get it to pass, I had to clean up the ITP database in this way for every iteration of the test. We should probably investigate why `_clearResourceLoadStatistics` isn't sufficient separately.